File: solutions/c/dnd-character/1/dnd_character.c
#include "dnd_character.h"
#include <math.h>
#include <time.h>
#include <stdlib.h>

int modifier(int score){
    return (int)floor(((float)score-10.0)/2.0);
}

int ability(void){
    srand(time(NULL));
    int dice_1 = 1+rand() % 5;
    int dice_2 = 1+rand() % 5;
    int dice_3 = 1+rand() % 5;
    int dice_4 = 1+rand() % 5;
    return dice_2+dice_3+dice_4-dice_1;
}

dnd_character_t make_dnd_character(void){
    dnd_character_t character;
    character.strength=ability();
    character.dexterity=ability();
    character.constitution=ability();
    character.intelligence=ability();
    character.wisdom=ability();
    character.charisma=ability();
    character.hitpoints=10+modifier(character.constitution);
    return character;
}

File: solutions/python/dnd-character/1/dnd_character.py
import random
import math

class Character:
    def __init__(self):
        self.strength=self.ability()
        self.dexterity=self.ability()
        self.constitution=self.ability()
        self.intelligence=self.ability()
        self.wisdom=self.ability()
        self.charisma=self.ability()
        self.hitpoints=10+1

    def ability(self):
        random.seed(110)
        roll=[random.randint(1,6),random.randint(1,6),random.randint(1,6),random.randint(1,6)] 
        roll.sort(reverse=True)
        return sum(roll[:3])

def modifier(value):
    return math.floor((value-10)/2)


    
    
